### DISCUSSION TITLE
Does titiler download and cache image assets for subsequent reqests?

### DISCUSSION THREAD

**Discussion by sairahul1526** [developmentseed/titiler#777] [Q&A]
Does titiler download and cache image assets for subsequent reqests or it gets the image from main source for every request?

**Comment by vincentsarago** [comment]
Hi @sairahul1526 

> Does titiler download and cache image assets for subsequent reqests or it gets the image from main source for every request?

There is no cache mechanism within TiTiler (except GDAL internal cache). We have an example in https://developmentseed.org/titiler/examples/code/tiler_with_cache/ to show how you can add one within titiler but we usually put TiTiler behind a CDN (like AWS cloud front) which serves as a cache layer

**Reply by sairahul1526** [reply]
Hey @vincentsarago 

Thanks for the swift reply.

I'm not talking about tiles regenerated by Titiler. I'm talking about the main band assets which Titiler downloads and processes it to generate titles. Does it cache those main source band assets, anywhere in memory?

**Reply by vincentsarago** [reply]
sorry @sairahul1526 for the misunderstanding 😓 

So as I said, GDAL does some kind of block caching but they mostly get deleted once we close the dataset. This is something I've tried to work on with https://github.com/vincentsarago/vsifile but sadly I don't have any band width to continue working on this at the moment (and there were couple blockers on rasterio side too)

**Reply by mradamcox** [reply]
@sairahul1526 @vincentsarago See also https://github.com/developmentseed/titiler/discussions/561, this GDAL caching was causing me some trouble with how I had (originally) set some things up.
